packages/qwik/src/qwikloader.ts

```diff
@@ -101,21 +114,27 @@ export const qwikLoader = (doc: Document, hasInitialized?: number) => {
             emitEvent('qerror', { importError: true, error, symbol, uri });
           }
         }
-        const previousCtx = (doc as any)[Q_CONTEXT];
+        if (!handler) {
+          const error = new Error('Handler not found for ' + url.href);
+          emitEvent('qerror', { error, href: url.href });
+          // break out of the loop if handler is not found
+          break;
+        }
+        const previousCtx = doc[Q_CONTEXT];
         if (element[isConnected]) {
           const eventData = { symbol, element, reqTime };
           try {
-            (doc as any)[Q_CONTEXT] = [element, ev, url];
+            doc[Q_CONTEXT] = [element, ev, url];
             isSync || emitEvent<QwikSymbolEvent>('qsymbol', eventData);
             const results = handler(ev, element);
             // only await if there is a promise returned
-            if (isPromise(results)) {
+            if (isPromise(results as Promise<unknown>)) {
```

Comment:
how come it didn't need a cast before?

Reply:
if (!handler) fixes this so we don't need to cast it anymore